### ISSUE TITLE
Malformed RFC 5646 language tag

### ISSUE THREAD

**Issue by svenkubiak** [dropbox/dropbox-sdk-java#56]
After updating to 2.0.4 I am seeing the following error when calling dbxClientV2.files().listFolder

```
com.dropbox.core.BadRequestException: HTTP header "Dropbox-API-User-Locale": Malformed RFC 5646 language tag: "de_DE": Invalid character at index 2: "_".  (RFC 5646 specifies "-" as the separator, not "_".)
	at com.dropbox.core.DbxRequestUtil.unexpectedStatus(DbxRequestUtil.java:274) ~[dropgallery.jar:?]
	at com.dropbox.core.v2.DbxRawClientV2$1.execute(DbxRawClientV2.java:106) ~[dropgallery.jar:?]
	at com.dropbox.core.v2.DbxRawClientV2.executeRetriable(DbxRawClientV2.java:251) ~[dropgallery.jar:?]
	at com.dropbox.core.v2.DbxRawClientV2.rpcStyle(DbxRawClientV2.java:96) ~[dropgallery.jar:?]
	at com.dropbox.core.v2.files.DbxUserFilesRequests.listFolder(DbxUserFilesRequests.java:629) ~[dropgallery.jar:?]
	at com.dropbox.core.v2.files.DbxUserFilesRequests.listFolder(DbxUserFilesRequests.java:656) ~[dropgallery.jar:?]
```

**Comment by krieb** [comment]
Hi @svenkubiak, this issue is currently being tracked and will be fixed in the next release. In the meantime, the recommended work-around is to set your locale as an IETF BCP 47 Language Tag. For the example above, that would be `de-DE`. If your locale is hardcoded, then just change it to `"de-DE"`. Otherwise, since Java 7, you can call `Locale.toLanguageTag(..)` to get a properly formatted IETF BCP 47 Language Tag:
```
Locale locale = ...;
config = new DbxRequestConfig("MyApp/1.0", locale.toLanguageTag());
```
If you are compiling with an older version of Java, then I recommend setting it using `DbxRequestConfig.Builder.withUserLocaleFrom(..)`, which will attempt to do a basic conversion for you:
```
Locale locale = ...;
config = DbxRequestConfig.newBuilder("MyApp/1.0")
    .withUserLocaleFrom(locale)
    .build();
```
Finally, if you just want to use the user's locale from their Dropbox preferences (e.g. what the user sees when they log into Dropbox), omit the locale entirely:
```
config = new DbxRequestConfig("MyApp/1.0");
```
or explicitly say so with the builder:
```
config = DbxRequestConfig.newBuilder("MyApp/1.0")
    .withUserLocaleFromPreferences()
    .build();
```

**Comment by krieb** [comment]
Should be fixed in **2.0.5**.
